### ISSUE TITLE
Expected Performence

### ISSUE THREAD

**Issue by Frostinassiky** [JJBOY/BMN-Boundary-Matching-Network#6]
Thanks for your work to implement BMN on PyTorch!

May I ask, what is the model performance that you get?

On my side, the **AUC is 64.5%** and **AR@100 is 73.5**. Please find the details below.

I did not change any parameters before running the code. Minor typos were fixed.

```
+ python main.py --module Evaluation                                                                                                                                        
[INIT] Loaded annotations from validation subset.                                                                                                     
        Number of ground truth instances: 7293                                                                                                                              
        Number of proposals: 472628                                                                                                                   
        Fixed threshold for tiou score: [0.5  0.55 0.6  0.65 0.7  0.75 0.8  0.85 0.9  0.95]                                                                                 
[RESULTS] Performance on ActivityNet proposal task.                                                                                                   
        Area Under the AR vs AN curve: 64.50481283422461%                                                                                                                   
/home/xum/anaconda3/envs/GTAL/lib/python3.7/site-packages/matplotlib/figure.py:98: MatplotlibDeprecationWarning:                                      
Adding an axes using the same arguments as a previous axes currently reuses the earlier instance.  In a future version,                                                     a new instance will always be created and returned.  Meanwhile, this warning can be suppressed, and the future behavior                               ensured, by passing a unique label to each axes instance.                                                                                                                   
  "Adding an axes using the same arguments as a previous axes "                                                                                       
AR@1 is          0.30287947346770877                                                                                                                                        
AR@5 is          0.43831070889894425                                                                                                                  
AR@10 is         0.5157136980666392                                                                                                                                         
AR@100 is        0.7352392705333882
```

**Comment by JJBOY** [comment]
Hi, please refer to this issue. #4

**Comment by Frostinassiky** [comment]
Ahaa, I see. Let's keep this issue open and wait for a better hyperparameters

**Comment by yklilfft** [comment]
Hello, I just read PaddlePaddle code the BMN author released, I found the details of model are not really same as paper. I tried to copy his hyperparameters in this pytorch version, but still cannot get performance like PaddlePaddle version( best AR@100 is about 74.5 ,AUC is nearly 65%). I notice that in his version, the BM map generation function is more complex, maybe it's the reason.


If u are familiar with PaddlePaddle framework, you can try to learn about this code:
https://github.com/PaddlePaddle/models/tree/develop/PaddleCV/PaddleVideo/models/bmn

**Comment by JJBOY** [comment]
@yklilfft  Well, thanks for your reminding me to look the code author provided. I will find the reason and update my code soon.

**Comment by JJBOY** [comment]
@Frostinassiky @yklilfft 
Thank for your attenstion. Now I has fix the differences between my codebase and the official's. And get very close results to the paper and official codebase. Please look at README to find the new results and more detials can be found in the new codes updated.

**Comment by Frostinassiky** [comment]
@JJBOY Great! thanks for the work!

**Comment by yklilfft** [comment]
Awesome! I have tried to update too, however I just only get about 74.8(AR@100), sadly…… Thanks a lot for your awesome update~

**Comment by JJBOY** [comment]
Really? Is the parameters totally same with me? That's strange. However, I dont save the model. I will run my code again and upload my model.

**Comment by Frostinassiky** [comment]
@JJBOY I got similar performance as @yklilfft did
```
train subset video numbers: 9649
validation subset video numbers: 4728
/opt/anaconda3/lib/python3.7/site-packages/torch/optim/lr_scheduler.py:82: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule.See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)
BMN training loss(epoch 0): tem_loss: 1.181, pem class_loss: 0.363, pem reg_loss: 0.020, total_loss: 1.747
BMN training loss(epoch 0): tem_loss: 1.164, pem class_loss: 0.357, pem reg_loss: 0.021, total_loss: 1.731
BMN training loss(epoch 1): tem_loss: 1.119, pem class_loss: 0.337, pem reg_loss: 0.018, total_loss: 1.639
BMN training loss(epoch 1): tem_loss: 1.147, pem class_loss: 0.331, pem reg_loss: 0.017, total_loss: 1.650
BMN training loss(epoch 2): tem_loss: 1.097, pem class_loss: 0.321, pem reg_loss: 0.016, total_loss: 1.583
BMN training loss(epoch 2): tem_loss: 1.139, pem class_loss: 0.335, pem reg_loss: 0.017, total_loss: 1.643
BMN training loss(epoch 3): tem_loss: 1.083, pem class_loss: 0.313, pem reg_loss: 0.016, total_loss: 1.554
BMN training loss(epoch 3): tem_loss: 1.130, pem class_loss: 0.324, pem reg_loss: 0.016, total_loss: 1.618
BMN training loss(epoch 4): tem_loss: 1.067, pem class_loss: 0.306, pem reg_loss: 0.015, total_loss: 1.526
BMN training loss(epoch 4): tem_loss: 1.130, pem class_loss: 0.335, pem reg_loss: 0.017, total_loss: 1.633
BMN training loss(epoch 5): tem_loss: 1.061, pem class_loss: 0.302, pem reg_loss: 0.015, total_loss: 1.513
BMN training loss(epoch 5): tem_loss: 1.136, pem class_loss: 0.330, pem reg_loss: 0.016, total_loss: 1.627
BMN training loss(epoch 6): tem_loss: 1.021, pem class_loss: 0.280, pem reg_loss: 0.014, total_loss: 1.437
BMN training loss(epoch 6): tem_loss: 1.125, pem class_loss: 0.320, pem reg_loss: 0.015, total_loss: 1.597
BMN training loss(epoch 7): tem_loss: 1.002, pem class_loss: 0.270, pem reg_loss: 0.013, total_loss: 1.404
BMN training loss(epoch 7): tem_loss: 1.127, pem class_loss: 0.325, pem reg_loss: 0.016, total_loss: 1.608
BMN training loss(epoch 8): tem_loss: 0.993, pem class_loss: 0.265, pem reg_loss: 0.013, total_loss: 1.388
BMN training loss(epoch 8): tem_loss: 1.137, pem class_loss: 0.332, pem reg_loss: 0.015, total_loss: 1.623
validation subset video numbers: 4728
Post processing start
Post processing finished
[INIT] Loaded annotations from validation subset.
        Number of ground truth instances: 7293
        Number of proposals: 472616
        Fixed threshold for tiou score: [0.5  0.55 0.6  0.65 0.7  0.75 0.8  0.85 0.9  0.95]
[RESULTS] Performance on ActivityNet proposal task.
        Area Under the AR vs AN curve: 67.21659810777457%
/home/xum/BMN-Boundary-Matching-Network/eval.py:52: MatplotlibDeprecationWarning: Adding an axes using the same arguments as a previous axes currently reuses the earlier instance.  In a future version, a new instance will always be created and returned.  Meanwhile, this warning can be suppressed, and the future behavior ensured, by passing a unique label to each axes instance.
  plt.setp(plt.axes().get_xticklabels(), fontsize=fn_size)
/home/xum/BMN-Boundary-Matching-Network/eval.py:53: MatplotlibDeprecationWarning: Adding an axes using the same arguments as a previous axes currently reuses the earlier instance.  In a future version, a new instance will always be created and returned.  Meanwhile, this warning can be suppressed, and the future behavior ensured, by passing a unique label to each axes instance.
  plt.setp(plt.axes().get_yticklabels(), fontsize=fn_size)
AR@1 is          0.33515699986288217
AR@5 is          0.49284245166598106
AR@10 is         0.565802824626354
AR@100 is        0.7508295625942685

```

**Comment by yklilfft** [comment]
BMN testing loss(epoch 8,iter 259/296): tem_loss: 1.130, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.607
BMN testing loss(epoch 8,iter 260/296): tem_loss: 1.130, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.607
BMN testing loss(epoch 8,iter 261/296): tem_loss: 1.130, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.607
BMN testing loss(epoch 8,iter 262/296): tem_loss: 1.131, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.607
BMN testing loss(epoch 8,iter 263/296): tem_loss: 1.130, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.607
BMN testing loss(epoch 8,iter 264/296): tem_loss: 1.131, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.607
BMN testing loss(epoch 8,iter 265/296): tem_loss: 1.131, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.608
BMN testing loss(epoch 8,iter 266/296): tem_loss: 1.132, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.609
BMN testing loss(epoch 8,iter 267/296): tem_loss: 1.131, pem class_loss: 0.324, pem reg_loss: 0.015, total_loss: 1.608
BMN testing loss(epoch 8,iter 268/296): tem_loss: 1.132, pem class_loss: 0.324, pem reg_loss: 0.015, total_loss: 1.608
BMN testing loss(epoch 8,iter 269/296): tem_loss: 1.133, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.610
BMN testing loss(epoch 8,iter 270/296): tem_loss: 1.133, pem class_loss: 0.326, pem reg_loss: 0.015, total_loss: 1.611
BMN testing loss(epoch 8,iter 271/296): tem_loss: 1.133, pem class_loss: 0.326, pem reg_loss: 0.015, total_loss: 1.611
BMN testing loss(epoch 8,iter 272/296): tem_loss: 1.133, pem class_loss: 0.326, pem reg_loss: 0.015, total_loss: 1.611
BMN testing loss(epoch 8,iter 273/296): tem_loss: 1.133, pem class_loss: 0.326, pem reg_loss: 0.015, total_loss: 1.612
BMN testing loss(epoch 8,iter 274/296): tem_loss: 1.132, pem class_loss: 0.326, pem reg_loss: 0.015, total_loss: 1.611
BMN testing loss(epoch 8,iter 275/296): tem_loss: 1.132, pem class_loss: 0.326, pem reg_loss: 0.015, total_loss: 1.611
BMN testing loss(epoch 8,iter 276/296): tem_loss: 1.132, pem class_loss: 0.326, pem reg_loss: 0.015, total_loss: 1.611
BMN testing loss(epoch 8,iter 277/296): tem_loss: 1.132, pem class_loss: 0.326, pem reg_loss: 0.015, total_loss: 1.612
BMN testing loss(epoch 8,iter 278/296): tem_loss: 1.132, pem class_loss: 0.326, pem reg_loss: 0.015, total_loss: 1.611
BMN testing loss(epoch 8,iter 279/296): tem_loss: 1.132, pem class_loss: 0.326, pem reg_loss: 0.015, total_loss: 1.611
BMN testing loss(epoch 8,iter 280/296): tem_loss: 1.132, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.610
BMN testing loss(epoch 8,iter 281/296): tem_loss: 1.132, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.610
BMN testing loss(epoch 8,iter 282/296): tem_loss: 1.131, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.609
BMN testing loss(epoch 8,iter 283/296): tem_loss: 1.132, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.610
BMN testing loss(epoch 8,iter 284/296): tem_loss: 1.131, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.609
BMN testing loss(epoch 8,iter 285/296): tem_loss: 1.131, pem class_loss: 0.326, pem reg_loss: 0.015, total_loss: 1.610
BMN testing loss(epoch 8,iter 286/296): tem_loss: 1.132, pem class_loss: 0.326, pem reg_loss: 0.015, total_loss: 1.611
BMN testing loss(epoch 8,iter 287/296): tem_loss: 1.132, pem class_loss: 0.326, pem reg_loss: 0.015, total_loss: 1.611
BMN testing loss(epoch 8,iter 288/296): tem_loss: 1.131, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.610
BMN testing loss(epoch 8,iter 289/296): tem_loss: 1.132, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.610
BMN testing loss(epoch 8,iter 290/296): tem_loss: 1.132, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.610
BMN testing loss(epoch 8,iter 291/296): tem_loss: 1.133, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.611
BMN testing loss(epoch 8,iter 292/296): tem_loss: 1.133, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.611
BMN testing loss(epoch 8,iter 293/296): tem_loss: 1.133, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.611
BMN testing loss(epoch 8,iter 294/296): tem_loss: 1.133, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.611
BMN testing loss(epoch 8,iter 295/296): tem_loss: 1.133, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.611
BMN testing loss(epoch 8): tem_loss: 1.133, pem class_loss: 0.325, pem reg_loss: 0.015, total_loss: 1.611
finish training process
validation subset video numbers: 4728
Post processing start
Post processing finished
[INIT] Loaded annotations from validation subset.
        Number of ground truth instances: 7293
        Number of proposals: 472616
        Fixed threshold for tiou score: [0.5  0.55 0.6  0.65 0.7  0.75 0.8  0.85 0.9  0.95]
[RESULTS] Performance on ActivityNet proposal task.
        Area Under the AR vs AN curve: 67.41598793363501%
/home/ykli/BMN-Boundary-Matching-Network/eval.py:52: MatplotlibDeprecationWarning: Adding an axes using the same arguments as a previous axes currently reuses the earlier instance.  In a future version, a new instance will always be created and returned.  Meanwhile, this warning can be suppressed, and the future behavior ensured, by passing a unique label to each axes instance.
  plt.setp(plt.axes().get_xticklabels(), fontsize=fn_size)
/home/ykli/BMN-Boundary-Matching-Network/eval.py:53: MatplotlibDeprecationWarning: Adding an axes using the same arguments as a previous axes currently reuses the earlier instance.  In a future version, a new instance will always be created and returned.  Meanwhile, this warning can be suppressed, and the future behavior ensured, by passing a unique label to each axes instance.
  plt.setp(plt.axes().get_yticklabels(), fontsize=fn_size)
AR@1 is          0.33555464143699437
AR@5 is          0.49625668449197863
AR@10 is         0.5687645687645688
AR@100 is        0.7536816125051419

I get the same result.
Process is the same, it doesn't influence the result. But when stride is set to num_sample, speed will be faster.

**Comment by lovelyczli** [comment]
Great!
